assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js

```diff
@@ -155,16 +154,18 @@ const WidgetAreaRenderer = ( { slug } ) => {
 		}
 
 		return classNames;
-	}, [ widgets, activeWidgets ] );
+	}, [ activeWidgets ] );
```

Comment:
@eugene-manuilov Good catch, `useMemo` is not relevant here because we're creating a new array on every render so we're unable to check dependencies based on referential equality. If we were able to abstract the `activeWidgets` filtering logic into a selector we'd be able to use the `useMemo` optimization as `useSelector` returns a cached result which allows for a referential equality check (returns the same reference if unchanged). Unfortunately how we check for a null render causes issues with hooks rendering order (throws error) so this is not an option.

Reply:
Yep, that's right. Thanks for updating it!